In [1]:
import pandas as pd
import numpy as np

In [2]:
experiment = pd.read_excel('experiment.xlsx')
costar = pd.read_excel('test 1.xlsx')
small = pd.read_excel('test2.xlsx')

#### Experiment Shape = (1150,25)
#### Costar Shape = (238,11)

In [3]:
costar

Company Name                      Address  \
0                            3 X P Inc         5850 Washington Blvd   
1                             7-Eleven    5900-5904 Washington Blvd   
2                       A&G Lumber Co.         5940 Washington Blvd   
3    AAA Tile & Stone Clearance Center  3221-3223 S La Cienega Blvd   
4                     ABC Internet Inc           3327 La Cienega Pl   
..                                 ...                          ...   
233                  Whitestone Design       5813 W Washington Blvd   
234                        Whole Foods       3321 S La Cienega Blvd   
235                       Writers Blok  2673-2677 S La Cienega Blvd   
236                 Zealot Productions       3201 S La Cienega Blvd   
237               Zevitas & Marcus LLC       2754 S La Cienega Blvd   

            City State    Zip       Ten_Phone  SF Occupied  \
0    Culver City    CA  90232             NaN            0   
1    Culver City    CA  90232  (310) 559-6919         2602   
2    Culver City    CA  90232  (310) 838-6222            0   
3    Los Angeles    CA  90016  (310) 559-5979            0   
4    Los Angeles    CA  90016             NaN            0   
..           ...   ...    ...             ...          ...   
233  Culver City    CA  90232  (323) 857-0222          500   
234  Los Angeles    CA  90016             NaN        50000   
235  Los Angeles    CA  90034             NaN         1300   
236  Los Angeles    CA  90016  (213) 537-1234          500   
237  Los Angeles    CA  90034             NaN         2000   

                     Location  # Emp     Submarket          Industry Type  
0             Single Location      0  Westside Ind  Public Administration  
1    Retail Store/Br Corp/LLC      7   Culver City               Retailer  
2             Single Location     10   Culver City               Retailer  
3             Single Location      5  Westside Ind             Wholesaler  
4             Single Location      0  Westside Ind            Information  
..                        ...    ...           ...                    ...  
233           Single Location      4   Culver City           Construction  
234  Retail Store/Br Corp/LLC      0   Culver City               Retailer  
235                       NaN      0  Westside Ind                    NaN  
236           Single Location      8   Culver City            Information  
237           Single Location      0  Westside Ind               Retailer  

[238 rows x 11 columns]

In [4]:
experiment.head()

New Additions Building Number Building Status Building Details Number  \
0             Y      BLDG105081          ON-NET       Enterprise  12777   
1             Y      BLDG168311          ON-NET       Enterprise   2001   
2             Y      BLDG150026          ON-NET       Enterprise   2227   
3             Y      BLDG169091          ON-NET       Enterprise  29315   
4             Y      BLDG167417          ON-NET       Enterprise    300   

  Prefix Street Name Type Suffix              Address  ...   Longitude  \
0    NaN  High Bluff   Dr    NaN  12777 High Bluff Dr  ... -117.239387   
1      E         4th   St    NaN        2001 E 4th St  ... -117.838500   
2    NaN  Enterprise   St    NaN   2227 Enterprise St  ... -117.116800   
3    NaN     Central  Ave    NaN    29315 Central Ave  ... -117.338300   
4    NaN        Sill   Rd    NaN          300 Sill Rd  ... -121.723000   

       CLLI                                  CBSA Lata  CoStar pull  \
0  SNDJCAD2                San Diego-Carlsbad, CA  732            Y   
1  SNAQCA07    Los Angeles-Long Beach-Anaheim, CA  730            Y   
2  ESCNCASL                San Diego-Carlsbad, CA  732            Y   
3  LKERCAGM  Riverside-San Bernardino-Ontario, CA  730            Y   
4  WTVLCAFU            Santa Cruz-Watsonville, CA  722         None   

   OSC Match Zoom match  AE Owner  AE owner date List Distributed?  
0        NaN        NaN       NaN            NaT               NaT  
1          Y          Y  David Ko     2022-05-01        2022-05-01  
2        NaN        NaN       NaN            NaT               NaT  
3        NaN        NaN       NaN            NaT               NaT  
4        NaN        NaN       NaN            NaT               NaT  

[5 rows x 25 columns]

In [5]:
costar.head()

Company Name                      Address  \
0                          3 X P Inc         5850 Washington Blvd   
1                           7-Eleven    5900-5904 Washington Blvd   
2                     A&G Lumber Co.         5940 Washington Blvd   
3  AAA Tile & Stone Clearance Center  3221-3223 S La Cienega Blvd   
4                   ABC Internet Inc           3327 La Cienega Pl   

          City State    Zip       Ten_Phone  SF Occupied  \
0  Culver City    CA  90232             NaN            0   
1  Culver City    CA  90232  (310) 559-6919         2602   
2  Culver City    CA  90232  (310) 838-6222            0   
3  Los Angeles    CA  90016  (310) 559-5979            0   
4  Los Angeles    CA  90016             NaN            0   

                   Location  # Emp     Submarket          Industry Type  
0           Single Location      0  Westside Ind  Public Administration  
1  Retail Store/Br Corp/LLC      7   Culver City               Retailer  
2           Single Location     10   Culver City               Retailer  
3           Single Location      5  Westside Ind             Wholesaler  
4           Single Location      0  Westside Ind            Information

In [6]:
experiment.columns

Index(['New Additions', 'Building Number', 'Building Status',
       'Building Details', 'Number', 'Prefix', 'Street Name', 'Type', 'Suffix',
       'Address', 'City', 'State', 'Zip', ' Tier', 'Latitude', 'Longitude',
       'CLLI', 'CBSA', 'Lata', 'CoStar pull', 'OSC Match', 'Zoom match',
       'AE Owner', 'AE owner date', 'List Distributed?'],
      dtype='object')

In [7]:
df = experiment[['Number','Prefix','Street Name','Type','Address',
                'City','State','Zip','CoStar pull','OSC Match',
                'Zoom match','AE Owner','AE owner date','List Distributed?']]
df.head()

Number Prefix Street Name Type              Address           City State  \
0  12777    NaN  High Bluff   Dr  12777 High Bluff Dr      San Diego    CA   
1   2001      E         4th   St        2001 E 4th St      Santa Ana    CA   
2   2227    NaN  Enterprise   St   2227 Enterprise St      Escondido    CA   
3  29315    NaN     Central  Ave    29315 Central Ave  Lake Elsinore    CA   
4    300    NaN        Sill   Rd          300 Sill Rd    Watsonville    CA   

     Zip CoStar pull OSC Match Zoom match  AE Owner AE owner date  \
0  92130           Y       NaN        NaN       NaN           NaT   
1  92705           Y         Y          Y  David Ko    2022-05-01   
2  92029           Y       NaN        NaN       NaN           NaT   
3  92532           Y       NaN        NaN       NaN           NaT   
4  95076        None       NaN        NaN       NaN           NaT   

  List Distributed?  
0               NaT  
1        2022-05-01  
2               NaT  
3               NaT  
4               NaT

In [8]:
df['CoStar pull'].value_counts()

Y       383
None     48
Name: CoStar pull, dtype: int64

#### As of 6/23/22, there are 383 buildings with a Costar Pull match, and 48 buildings that did not show a matching Costar Pull. So we will look at the buildings that we have not looked up on Costar yet, which is 719 buildings.

In [9]:
df = df[df['CoStar pull'].isna()]
df.shape

(719, 14)

In [10]:
df['Full Address'] = df['Address'] +' '+ df['Zip']
df.head()

Number Prefix      Street Name  Type                  Address  \
90  11120    NaN          Norwalk  Blvd       11120 Norwalk Blvd   
93  11149      N     Torrey Pines    Rd  11149 N Torrey Pines Rd   
94  11165    NaN        Sepulveda  Blvd     11165 Sepulveda Blvd   
96  11188    NaN           Citrus   Ave         11188 Citrus Ave   
98    112    NaN  Lakeview Canyon    Rd   112 Lakeview Canyon Rd   

                City State    Zip CoStar pull OSC Match Zoom match AE Owner  \
90  Santa Fe Springs    CA  90670         NaN       NaN        NaN      NaN   
93         San Diego    CA  92121         NaN       NaN        NaN      NaN   
94     Mission Hills    CA  91345         NaN       NaN        NaN      NaN   
96           Fontana    CA  92337         NaN       NaN        NaN      NaN   
98     Thousand Oaks    CA  91362         NaN       NaN        NaN      NaN   

   AE owner date List Distributed?                   Full Address  
90           NaT               NaT       11120 Norwalk Blvd 90670  
93           NaT               NaT  11149 N Torrey Pines Rd 92121  
94           NaT               NaT     11165 Sepulveda Blvd 91345  
96           NaT               NaT         11188 Citrus Ave 92337  
98           NaT               NaT   112 Lakeview Canyon Rd 91362

In [11]:
small.head()

Company Name                      Address  \
0  AAA Tile & Stone Clearance Center  3221-3223 S La Cienega Blvd   
1                            Anagram       3137 S La Cienega Blvd   
2                    Bolt/Mothership       3211 S La Cienega Blvd   
3                     Cavendish Grey  3221-3223 S La Cienega Blvd   
4                            Chevron       3300 S La Cienega Blvd   

          City State    Zip       Ten_Phone  SF Occupied  \
0  Los Angeles    CA  90016  (310) 559-5979            0   
1  Los Angeles    CA  90016             NaN         6686   
2  Los Angeles    CA  90016             NaN         5000   
3  Los Angeles    CA  90016  (323) 653-2230          500   
4  Los Angeles    CA  90016  (323) 935-9350          596   

                   Location  # Emp     Submarket  \
0           Single Location      5  Westside Ind   
1          Main HQ Corp/LLC    100   Culver City   
2                       NaN      0   Culver City   
3  Single Location Corp/LLC      0  Westside Ind   
4  Retail Store/Br Corp/LLC      4   Culver City   

                                      Industry Type  
0                                        Wholesaler  
1  Professional, Scientific, and Technical Services  
2                                               NaN  
3  Professional, Scientific, and Technical Services  
4                                          Retailer

#### We change the Costar data to show only the rows with matching On-Net Buildings

In [12]:
match = small[(small['Address'] +' '+ small['Zip'].map(str)).isin(df['Full Address'])]
match

Company Name                 Address         City State    Zip  \
2  Bolt/Mothership  3211 S La Cienega Blvd  Los Angeles    CA  90016   

  Ten_Phone  SF Occupied Location  # Emp    Submarket Industry Type  
2       NaN         5000      NaN      0  Culver City           NaN

#### We want to append the 'match' table to a master DF with all the finished data 

In [13]:
master = pd.DataFrame(columns = costar.columns)

In [14]:
master = master.append(match)

C:\Users\melee\AppData\Local\Temp\ipykernel_16260\4275024012.py:1: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  master = master.append(match)


In [15]:
master

Company Name                 Address         City State    Zip  \
2  Bolt/Mothership  3211 S La Cienega Blvd  Los Angeles    CA  90016   

  Ten_Phone SF Occupied Location # Emp    Submarket Industry Type  
2       NaN        5000      NaN     0  Culver City           NaN

#### We want to add a 'Y' to df['Costar pull']

In [16]:
master_list = (master['Address'] +' '+ master['Zip'].map(str)).tolist()
master_list

['3211 S La Cienega Blvd 90016']

In [17]:
df['Costar pull'] = np.where(df['Full Address'].isin(master_list), 'Y', np.NaN)

In [18]:
df['Costar pull'].value_counts()

nan    718
Y        1
Name: Costar pull, dtype: int64